cpp/src/arrow/compute/exec/expression.cc

```diff
@@ -42,18 +42,17 @@ using internal::checked_pointer_cast;
 
 namespace compute {
 
-Expression::Expression(Call call) : impl_(std::make_shared<Impl>(std::move(call))) {}
+Expression::Expression(Call&& call) : impl_(std::make_shared<Impl>(call)) {}
```

Comment:
A declaration of `Call&& call` doesn't get automatically moved on the first reference expression, so we still need the call to `std::move`
```suggestion
Expression::Expression(Call&& call) : impl_(std::make_shared<Impl>(std::move(call))) {}
```